plugins/ipam/dhcp/options.go

```diff
@@ -40,6 +42,45 @@ func classfulSubnet(sn net.IP) net.IPNet {
 	}
 }
 
+func parseDNSServers(opts dhcp4.Options) []string {
+	// Parse DNS from dhcp4 options packet
+	// ns = 4 bytes, net.IP
+	nameservers := []string{}
+	var ns string
+	if opt, ok := opts[dhcp4.OptionDomainNameServer]; ok {
+		for len(opt) >= 4 {
+			ns = net.IP(opt[0:4]).String()
+			nameservers = append(nameservers, ns)
+			opt = opt[4:]
+		}
+
+	}
+
+	return nameservers
+}
+
+func parseDNSDomain(opts dhcp4.Options) string {
+	// Parse domainname option
+	var domainname string
+	// Thanks to: https://github.com/asaskevich/govalidator
+	var DNSName string = `^([a-zA-Z0-9]{1}[a-zA-Z0-9_-]{1,62}){1}(\.[a-zA-Z0-9]{1}[a-zA-Z0-9_-]{1,62})*$`
+	if opt, ok := opts[dhcp4.OptionDomainName]; ok {
+		domainname = string(opt[:])
+		// RFC 1035 section 2.3.4 and 3.1
+		if len(strings.Replace(domainname, ".", "", -1)) > 255 {
```

Comment:
Actually, it could be 254 (and after adding a single letter hostname, the result will be 255 bytes), which means that the regex is wrong (my bad), the domain "a.com" is valid, but the regex will fail validating this name (extra test on this, including dash, which are only accepted after a letter or digit).